samples/apps/research-assistant-react/backend/message_handler.py

```diff
@@ -0,0 +1,165 @@
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
+import json
+
+import autogen
+from autogen import oai
+from autogen import Agent
+from autogen.code_utils import extract_code
+
+from raworkflow.raworkflow_helper import print_messages
+from raworkflow import RAWorkflow
+from raworkflow.ratwoagentworkflow import TwoAgentWorkflow
+from raworkflow.raplannerworkflow import HumanInLoopPlannerExecutor
+
+from utils.code_utils import extract_code_result, learn_skill, utils_2_prompt
+
+import utils.utils
+from utils import generate_oai_reply, copy_utils
+
+from enum import Enum
+
+
+class AgentWorkFlow(Enum):
+    CODER_ONLY = "TwoAgents"
+    PLANNER_CODER = "PlannerCoder"
+
+
+# Sorted listed of available RA types
+AVAILABLE_RAS = [
+    {
+        "agent": AgentWorkFlow.CODER_ONLY,
+        "title": "Coder Only",
+        "description": "A two agent workflow with a coder and an assistant",
+    },
+    # {
```

Comment:
Why are these lines commented out?

Reply:
Because we only demonstrate the Coder workflow.

Reply:
Hmm, but the code contains files for the planner workflow? Commented lines without any context can cause confusion.

Reply:
@gagb, the planner code never got tested. This code is a sample and as is. We could add a comment saying this code demonstrate a planner workflow.

Reply:
Either more testing + uncomment or no testing + more clarifying documentation is okay with me. Both reduce confusion.